azurerm/resource_arm_sql_failover_group.go

```diff
@@ -0,0 +1,357 @@
+			return tf.ImportAsExistsError("azurerm_sql_failover_group", *existing.ID)
+		}
+	}
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	properties := sql.FailoverGroup{
+		FailoverGroupProperties: &sql.FailoverGroupProperties{
+			ReadOnlyEndpoint:  expandSqlFailoverGroupReadOnlyPolicy(d),
+			ReadWriteEndpoint: expandSqlFailoverGroupReadWritePolicy(d),
+			PartnerServers:    expandSqlFailoverGroupPartnerServers(d),
+		},
+		Tags: expandTags(tags),
+	}
+
+	if r, ok := d.Get("databases").(*schema.Set); ok && r.Len() > 0 {
+		var databases []string
+		for _, v := range r.List() {
+			s := v.(string)
+			databases = append(databases, s)
+		}
+
+		properties.Databases = &databases
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, serverName, name, properties)
+	if err != nil {
+		return fmt.Errorf("Error issuing create/update request for SQL Failover Group %q (Resource Group %q, Server %q): %+v", name, resourceGroup, serverName, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting on create/update future for SQL Failover Group %q (Resource Group %q, Server %q): %+v", name, resourceGroup, serverName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, serverName, name)
+	if err != nil {
+		return fmt.Errorf("Error issuing get request for SQL Failover Group %q (Resource Group %q, Server %q): %+v", name, resourceGroup, serverName, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmSqlFailoverGroupRead(d, meta)
+}
+
+func resourceArmSqlFailoverGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).sqlFailoverGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	serverName := id.Path["servers"]
+	name := id.Path["failoverGroups"]
+
+	resp, err := client.Get(ctx, resourceGroup, serverName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request for SQL Failover Group %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("server_name", serverName)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := resp.FailoverGroupProperties; props != nil {
+		if err := d.Set("read_write_endpoint_failover_policy", flattenSqlFailoverGroupReadWritePolicy(props.ReadWriteEndpoint)); err != nil {
+			return fmt.Errorf("Error setting `read_write_endpoint_failover_policy`: %+v", err)
+		}
+
+		if err := d.Set("readonly_endpoint_failover_policy", flattenSqlFailoverGroupReadOnlyPolicy(props.ReadOnlyEndpoint)); err != nil {
+			return fmt.Errorf("Error setting `read_only_endpoint_failover_policy`: %+v", err)
+		}
+
+		d.Set("databases", set.FromStringSlice(*props.Databases))
```

Comment:
We will need to nil check this:
```suggestion
		if props.Databases != nil {
			d.Set("databases", set.FromStringSlice(*props.Databases))
		}
```